scripts/docker/CC-Devel/buildcc.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+set -e
+
+function usage() {
+    echo "${0}"
+    echo "  -h  Print this usage information. Optional."
+    echo "  -s  Directory of CodeCompass source. If not specified then"
+    echo "      CC_SOURCE environment variable will be used. Any of them"
+    echo "      is mandatory."
+    echo "  -o  Directory of generated output artifacts. If not specified then"
+    echo "      CC_BUILD environment variable will be used. Any of them"
+    echo "      is mandatory."
+}
+
+cc_source_dir="${CC_SOURCE}"
+cc_output_dir="${CC_BUILD}"
+while getopts ":hs:o:" option; do
+    case ${option} in
+        h)
+            usage
+            exit 0
+            ;;
+        s)
+            cc_source_dir="${OPTARG}"
+            ;;
+        o)
+            cc_output_dir="${OPTARG}"
+            ;;
+        *)
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+if [[ -z "${cc_source_dir}" ]]; then
+    echo "CodeCompass source directory should be defined." >&2
+    usage
+    exit 3
+fi
+
+if [[ -z "${cc_output_dir}" ]]; then
+    echo "Output directory of build should be defined." >&2
+    usage
+    exit 4
+fi
+
+developer_id="$(id --user)"
+developer_group="$(id --group)"
+
+if [[ "$developer_id" -eq 0 ]] || [[ "$developer_group" -eq 0 ]]; then
+    echo "'${0}' should not run as root." >&2
+    exit 2
+fi
+
+mkdir -p ${cc_output_dir}
+cc_source_mounted="/mnt/cc_source"
+cc_output_mounted="/mnt/cc_output"
+
+docker_command=("docker" "run" "--rm"                                       \
+  "--user=${developer_id}:${developer_group}"                               \
+  "--mount" "type=bind,source=${cc_source_dir},target=${cc_source_mounted}" \
```

Comment:
Using `Docker version 1.13.1, build 092cba3` (Ubuntu 16.04.1 LTS) the `mount` option does not exist.

```
12:36 $ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:        16.04
Codename:       xenial

12:36 $ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:        16.04
Codename:       xenial

12:36 $ sudo docker version
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64
 Experimental: false

12:38 $ docker run --help
[...]
      --memory-swap string                    Swap limit equal to memory plus swap: '-1' to enable unlimited swap
      --memory-swappiness int                 Tune container memory swappiness (0 to 100) (default -1)
      --name string                           Assign a name to the container
      --network string                        Connect a container to a network (default "default")
[...]
```

Reply:
Script uses canonical paths now.